infra/RepoManager.py

```diff
@@ -0,0 +1,220 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to manage a git repository via python.
+
+This class is to be used to implement git commands over
+a python API and manage the current state of the git repo.
+
+  Typical usage example:
+
+    r_man =  RepoManager('https://github.com/google/oss-fuzz.git')
+    r_man.checkout('5668cc422c2c92d38a370545d3591039fb5bb8d4')
+    r_man.getBranch('bisector')
+    r_man.close()
+"""
+import os
+import shutil
+import subprocess
+
+
+class RepoManagerException(Exception):
+  """Class to describe the exceptions in RepoManager."""
+
+  def __init__(self, message):
+    """ Init the exception class.
+
+    Args:
+      message: the message to propigate to user
+    """
+    super().__init__(message)
+
+
+class RepoManager(object):
+  """Class to manage git repos from python.
+
+  Attributes:
+    repo_url: The location of the git repo
+    local_dir: The location of where the repo clone is stored locally
+    repo_name: The name of the github project
+    repo_dir: The location of the main repo
+    full_path: The full filepath location of the main repo
+  """
+  repo_url = ''
+  repo_name = ''
+  repo_dir = ''
+  local_dir = ''
+  full_path = ''
+
+  def __init__(self, repo_url, commit=None, local_dir='tmp'):
+    """Constructs a repo manager class.
+
+    Args:
+      repo_url: The github url needed to clone
+      commit: The specified commit to be checked out
+      local_dir: The local location the repo will live in
+    """
+
+    self.repo_url = repo_url
+    self.local_dir = local_dir
+    self.repo_name = self.repo_url.split('/')[-1].strip('.git')
+    self.repo_dir = os.path.join(self.local_dir, self.repo_name)
+    self.full_path = os.path.join(os.getcwd(), self.repo_dir)
+    self._clone()
+    if branch is not None:
+      self.checkout(branch)
+
+    if commit is not None:
+      self.checkout_commit(commit)
+
+  def _clone(self):
+    """Creates a clone of the repo in the specified directory.
+
+      Raises:
+        RepoManagerException if the repo was not able to be cloned
+    """
+    if not os.path.exists(self.local_dir):
+      os.makedirs(self.local_dir)
+    self.remove_repo()
+    _, err = self._run_command(['git', 'clone', self.repo_url], self.local_dir)
+    if err is not None:
+      raise RepoManagerException(
+          'Failed cloning repo %s, with error %s)' % (self.repo_url, err))
+    if not self._is_git_repo():
+      raise RepoManagerException('%s is not a git repo' % self.repo_url)
+
+  def _run_command(self, command, location='.'):
+    """ Runs a shell command in the specified directory location.
+
+    Args:
+      command: The command as a list to be run
+
+    Returns:
+      The stdout of the command, the stderr of the command
+    """
+    cur_dir = os.getcwd()
+    os.chdir(location)
```

Comment:
I think `Popen` has `cwd` argument, better pass the directory using it instead of changing the current workdir explicitly: https://docs.python.org/2/library/subprocess.html#popen-constructor